carapax-i18n/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+//! An i18n handler for carapax
+
+#![warn(missing_docs)]
+
+#[cfg(feature = "gettext")]
+mod gettext;
+#[cfg(feature = "gettext")]
+pub use gettext::*;
+
+#[cfg(feature = "fluent")]
+#[macro_use]
+extern crate rental;
+#[cfg(feature = "fluent")]
+mod fluent;
+#[cfg(feature = "fluent")]
+pub use fluent::*;
+
+use carapax::{core::types::Update, Context, HandlerFuture, HandlerResult, UpdateHandler};
+use std::collections::HashMap;
+
+/// An i18n handler for carapax
+#[derive(Debug, Clone)]
+pub struct I18nHandler<R, T> {
+    resolver: R,
+    default: T,
+    languages: Vec<T>,
+}
+
+impl<R, T> I18nHandler<R, T> {
+    /// Creates new I18nHandler
+    ///
+    /// Search translator in languages or use default one
+    pub fn new<I>(resolver: R, default: T, languages: I) -> Self
+    where
+        I: IntoIterator<Item = T>,
+    {
+        Self {
+            resolver,
+            default,
+            languages: languages.into_iter().collect(),
+        }
+    }
+}
+
+impl<R, T> UpdateHandler for I18nHandler<R, T>
+where
+    R: LocaleResolver,
+    T: Translator + Send + Sync + 'static,
+{
+    fn handle(&self, context: &mut Context, update: &Update) -> HandlerFuture {
+        let lang = self.resolver.resolve(update);
+        let translator = self
```

Comment:
Why we cant use a `HashMap<Locale, Translator>`?

Reply:
+1